### PR TITLE
Add support for multiple UnexpectedSegmentBehaviour options

### PR DESCRIPTION
**Pull request by milkshakeuk**
Port `UnexpectedSegmentBehaviour` Options from `hapi`.

### CODE REVIEW

**Comment by milkshakeuk** [comment]
@IanGralinski-Invetech @AMCN41R could you review this if you get time?

**Comment by milkshakeuk** [comment]
@IanGralinski-Invetech, @AMCN41R do you think you might get chance before Xmas to review this pr?

**Comment by IanGralinski-Invetech** [comment]
@milkshakeuk I can take a look after Christmas before the end of the year.

File: src/NHapi.Base/Parser/MessageIterator.cs [hunk 1]
```diff
@@ -233,26 +250,23 @@ private void AddNonStandardSegmentAtCurrentPosition()
...
+            switch (parserOptions.UnexpectedSegmentBehaviour)
+            {
+                case UnexpectedSegmentBehaviour.DropToRoot:
+                    parentDefinitionPath = new List<Position>(currentDefinitionPath.GetRange(0, 1));
+                    parentStructure = this.message;
+                    currentDefinitionPath = currentDefinitionPath.GetRange(0, 2);
+                    break;
+                case UnexpectedSegmentBehaviour.ThrowHl7Exception:
+                    throw new HL7Exception($"Found unknown segment: {direction}");
+                default:
```
Comment (AMCN41R) [inline src/NHapi.Base/Parser/MessageIterator.cs:265]: commenting without context here, but...

is this truely the default behaviour? ie: if I added another value to `UnexpectedSegmentBehaviour` would I want this behaviour? If not, would suggest being expicit and including `case: UnexpectedSegmentBehaviour.AddInline` and then throwning an exception as the default case (`UnhandledUnexpectedSegmentBehaviourException` or something)

**Review by AMCN41R — APPROVED** [review]

File: src/NHapi.Base/Parser/MessageIterator.cs (hunk 1 above)
Comment (milkshakeuk) [reply src/NHapi.Base/Parser/MessageIterator.cs:265]: @AMCN41R also that was the default behaviour before the config option as you can see in the diff. all I did here was make it implicitly default (ported from hapi).

Comment (milkshakeuk) [reply src/NHapi.Base/Parser/MessageIterator.cs:265]: @AMCN41R I see what you are saying having thought about it, I will make the change.

Comment (milkshakeuk) [reply src/NHapi.Base/Parser/MessageIterator.cs:257]: @AMCN41R I had made this change.

**Review by AMCN41R — APPROVED** [review]

**Review by PhantomGrazzler — APPROVED** [review]

**Pull request merged**